app/src/main/java/com/onboarding/parkingsystemjava/mvp/model/ReserveModel.java

```diff
@@ -49,8 +52,12 @@ else if (reservation.getParkingLot() == ConstantUtils.INT_MINUS_ONE)
             return MISSING_LOT;
         else if (reservation.getUserPassword().isEmpty())
             return MISSING_PASSWORD;
-        else
+        else if (verifier.isReservationOK(this.reservation)) {
```

Comment:
pls use guard clauses

Reply:
https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

Reply:
i think this checker could also check all previous checks

Reply:
done, replaced if/else nodes with guard clauses and moved fields check to ReservationVerifier